Sources/PackageModel/IdentityResolver.swift

```diff
@@ -20,6 +21,10 @@ public protocol IdentityResolver {
     func resolveIdentity(for path: AbsolutePath) throws -> PackageIdentity
     func mappedLocation(for location: String) -> String
     func mappedIdentity(for identity: PackageIdentity) throws -> PackageIdentity
+
+    func mappedDependency(for dependency: PackageDependency, fileSystem: FileSystem) throws -> PackageDependency
```

Comment:
not convinced if adding all these to identity resolver is the right "place", maybe a separate utility?

Reply:
I mean, the only reason for the existence of identity resolver are actually these dependency mapping tasks. I'm not sure it makes sense to continue obscuring that by separating these two very related concerns into separate types. I could see _renaming_ identity resolver and also removing most of the API surface since it should no longer be necessary now that this type handles the process.

Reply:
the way I see it, `IdentityResolver` is a parser of identity from various sources. the mapping is next level, so maybe it needs to be hidden as an implementation detail of the higher level utility but I think its worth keeping the "parser" as a testable unit without the higher level Apis

Reply:
Ah yah, that sounds reasonable to me.

Reply:
Hm, various test utilities (e.g. `MockDependency`) have their very own notion of "mapping" as well, this would really become an even more sprawling change if we took away `IdentityResolver` from the public interface.

All of the tests really need to be cleaned up at some point so that they actually test the "real world" and not all kinds of fake scenarios that are just held up by workarounds in the production code, but I am not sure I want to take that on for a simple bug fix.

Which I guess means that the path of least resistance is to make a new type and pass it around in addition to the identity resolver where it is needed 😞

Reply:
Split these up into two types now, everything that needs both takes two arguments.